tests/gold_tests/remap/remap_reload.test.py

```diff
@@ -0,0 +1,88 @@
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import os
+
+Test.Summary = '''
+Test remap reloading
+'''
+Test.testName = 'remap_reload'
+
+replay_file_1 = "reload_1.replay.yaml"
+replay_file_2 = "reload_2.replay.yaml"
+replay_file_3 = "reload_3.replay.yaml"
+replay_file_4 = "reload_4.replay.yaml"
+
+tm = Test.MakeATSProcess("tm", command="traffic_manager", select_ports=True)
+tm.Disk.diags_log.Content = Testers.ContainsExpression("remap.config failed to load", "Remap should fail to load")
+remap_cfg_path = os.path.join(tm.Variables.CONFIGDIR, 'remap.config')
+
+pv = Test.MakeVerifierServerProcess("pv", "reload_server.replay.yaml")
+pv_port = pv.Variables.http_port
+tm.Disk.remap_config.AddLines([f"map http://alpha.ex http://alpha.ex:{pv_port}",
+                               f"map http://bravo.ex http://bravo.ex:{pv_port}",
+                               f"map http://charlie.ex http://charlie.ex:{pv_port}",
+                               f"map http://delta.ex http://delta.ex:{pv_port}"])
+tm.Disk.records_config.update({'proxy.config.url_remap.min_rules_required': 3})
+
+nameserver = Test.MakeDNServer("dns", default='127.0.0.1')
+tm.Disk.records_config.update({
+    'proxy.config.dns.nameservers': f"127.0.0.1:{nameserver.Variables.Port}", 'proxy.config.dns.resolv_conf': 'NULL'
+})
+
+# tm.Disk.records_config.update({
+#    'proxy.config.diags.debug.enabled': 1,
+#    'proxy.config.diags.debug.tags': 'http|dns|hostdb',
+# })
```

Comment:
Probably remove this commented out code?

Reply:
I debated that - it's handy if the test goes wrong, but I suppose it's easy enough to clone when needed.

Reply:
Oh, I see. I'm fine leaving it in. Please just add a comment saying something like: `# Uncomment for diagnostic logging.`